src/Microsoft.AspNet.OData.Shared/Routing/Conventions/RequestMethodExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.OData.Interfaces;
+
+namespace Microsoft.AspNet.OData.Routing.Conventions
+{
+    internal static class RequestMethodExtensions 
+    {
+        internal static ODataRequestMethod GetRequestMethodOrPreflightMethod(this IWebApiRequestMessage request)
+        {
+            if (request.Method == ODataRequestMethod.Options)
+            {
+                IEnumerable<string> values;
+                request.Headers.TryGetValues("Access-Control-Request-Method", out values);
```

Comment:
@marabooy According to the docs, `TryGetValues` returns true if the specified header and specified values are stored are stored in the `HttpHeaders` collection: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.trygetvalues?view=netcore-3.1
For that reason, I think you could rewrite this section as:
```c#
IEnumerable<string> values;
if (request.Headers.TryGetValues("Access-Control-Request-Method", out values)
{
    return ToODataRequestMethod(values.FirstOrDefault());
}
else
{
    return ODataRequestMethod.Unknown;
}
```
Note that the `TryGetValues()` method won't take a `List<string>` or even `IList<string>` out parameter - I tried it out.
In addition, you could use `.First()` (on the strength of the docs) but I think it may be safer to use `.FirstOrDefault()` and do a null-check first in the private `ToODataRequestMethod` helper method

Reply:
👍